code/cycler/src/wattrex_battery_cycler/mid/mid_dabs/mid_dabs.py

```diff
@@ -0,0 +1,393 @@
+        if CyclerDataDeviceTypeE.EPC in self.device_type:
+            if isinstance(ls_curr, tuple):
+                try:
+                    self.epc.set_ls_curr_limit(ls_curr[0], ls_curr[1])
+                except Exception as err:
+                    log.error(f"Error while setting ls current limit: {err}")
+                    raise Exception("Error while setting wait mode") from err #pylint: disable= broad-exception-raised
+            if isinstance(ls_volt, tuple):
+                try:
+                    self.epc.set_ls_volt_limit(ls_volt[0], ls_volt[1])
+                except Exception as err:
+                    log.error(f"Error while setting ls current limit: {err}")
+                    raise Exception("Error while setting wait mode") from err #pylint: disable= broad-exception-raised
+            if isinstance(ls_pwr, tuple):
+                try:
+                    self.epc.set_ls_pwr_limit(ls_pwr[0], ls_pwr[1])
+                except Exception as err:
+                    log.error(f"Error while setting ls current limit: {err}")
+                    raise Exception("Error while setting wait mode") from err #pylint: disable= broad-exception-raised
+            if isinstance(hs_volt, tuple):
+                try:
+                    self.epc.set_hs_volt_limit(hs_volt[0], hs_volt[1])
+                except Exception as err:
+                    log.error(f"Error while setting ls current limit: {err}")
+                    raise Exception("Error while setting wait mode") from err #pylint: disable= broad-exception-raised
+            if isinstance(temp, tuple):
+                try:
+                    self.epc.set_temp_limit(temp[0], temp[1])
+                except Exception as err:
+                    log.error(f"Error while setting ls current limit: {err}")
+                    raise Exception("Error while setting wait mode") from err #pylint: disable= broad-exception-raised
+        else:
+            log.error("The limits can not be change in this device")
+
+    def disable(self) -> None:
+        """Disable the devices.
+        """
+        try:
+            if CyclerDataDeviceTypeE.BISOURCE in self.device_type:
+                self.bisource.disable()
+            elif (CyclerDataDeviceTypeE.SOURCE in self.device_type and
+                CyclerDataDeviceTypeE.LOAD in self.device_type):
+                self.source.disable()
+                self.load.disable()
+            elif CyclerDataDeviceTypeE.EPC in self.device_type:
+                log.info("Disabling epc")
+                self.epc.disable()
+            else:
+                log.error("The device can not be disable")
+                raise ValueError("The device can not be disable")
+        except Exception as err:
+            log.error(f"Error while disabling device: {err}")
+            raise Exception("Error while disabling device") from err #pylint: disable= broad-exception-raised
+
+def __prepare_link_conf(link_conf: dict) -> dict:
```

Comment:
Remove it an perform checks on DataLink class initialization

Reply:
Done